### ISSUE TITLE
Chrome crashed

### ISSUE THREAD

**Issue by poolwar** [sitespeedio/sitespeed.io#1276]
[2016-10-28 02:54:30] Versions OS: linux 3.10.0-327.el7.x86_64 nodejs: v6.9.1 sitespeed.io: 4.0.0 browsertime: 1.0.0-beta.9 coach: 0.28.4
[2016-10-28 02:54:31] Starting chrome for analysing https://www.xiaoying.com 3 time(s)
[2016-10-28 02:54:31] Testing url https://www.xiaoying.com run 1
[2016-10-28 02:55:31] Browser failed to start in time, trying one more time.
_**[2016-10-28 02:56:31] https://www.xiaoying.com generated the following error in Browsertime BrowserError: unknown error: Chrome failed to start: crashed**_
  (Driver info: chromedriver=2.24.417424 (c5c5ea873213ee72e3d0929b47482681555340c3),platform=Linux 3.10.0-327.el7.x86_64 x86_64)
  (Driver info: chromedriver=2.24.417424 (c5c5ea873213ee72e3d0929b47482681555340c3),platform=Linux 3.10.0-327.el7.x86_64 x86_64)
    at BrowsertimeError (/usr/src/app/node_modules/browsertime/lib/support/errors.js:5:5)
    at BrowserError (/usr/src/app/node_modules/browsertime/lib/support/errors.js:13:5)
    at startBrowser.call.catch.tap.tap.catch.catch (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:79:15)
From previous event:
    at SeleniumRunner.start (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:78:13)
    at runIteration (/usr/src/app/node_modules/browsertime/lib/core/engine.js:167:21)
    at Promise.reduce (/usr/src/app/node_modules/browsertime/lib/core/engine.js:242:25)
From previous event:
    at Promise.resolve.tap.tap.tap (/usr/src/app/node_modules/browsertime/lib/core/engine.js:241:17)
From previous event:
    at Engine.run (/usr/src/app/node_modules/browsertime/lib/core/engine.js:240:8)
    at engine.start.then (/usr/src/app/lib/plugins/browsertime/analyzer.js:76:26)
From previous event:
    at Object.analyzeUrl (/usr/src/app/lib/plugins/browsertime/analyzer.js:76:8)
    at Object.processMessage (/usr/src/app/lib/plugins/browsertime/index.js:128:27)
    at queue.process (/usr/src/app/lib/support/queueHandler.js:157:57)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:92:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:73:84)
    at runCallback (timers.js:637:20)
    at tryOnImmediate (timers.js:610:5)
    at processImmediate [as _immediateCallback](timers.js:582:5)
    at SeleniumRunner.startBrowser (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:37:10)
    at SeleniumRunner.start (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:42:25)
    at runIteration (/usr/src/app/node_modules/browsertime/lib/core/engine.js:167:21)
    at Promise.reduce (/usr/src/app/node_modules/browsertime/lib/core/engine.js:242:25)
[2016-10-28 02:56:31] Render HTML for 1 page(s) 
[2016-10-28 02:56:32] HTML stored in /sitespeed.io/sitespeed-result/www.xiaoying.com/2016-10-28-02-54-30
[2016-10-28 02:56:32] Finished analysing https://www.xiaoying.com

how could i fix this problem？update chrome？thx

**Comment by soulgalore** [comment]
Hey @poolwar 
seen problems with Chrome on some OS for some pages. We should report it to the Chrome/Chromium team. What OS do use and how do you run sitespeed.io, using Docker or nodejs?

Best
Peter

**Comment by kikyoar** [comment]
the same issue
 **Versions OS: linux 4.4.0-21-generic nodejs: v6.9.1 sitespeed.io: 4.0.2 browsertime: 1.0.0-beta.9 coach: 0.28.4**

**Comment by soulgalore** [comment]
hey @kikyoar 
thanks for reporting. Is it a specfic URL? Would be great if you can share so we can isses for Chrome. I've seen URL working on OS X but not on Ubuntu etc.

Best
Peter

**Comment by kikyoar** [comment]
@soulgalore，thanks for your reply！
This is issue：

root@ubuntu:~# docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:4.0.2 --preScript aa.js -b chrome http://www.ctnma.cn/ioop-bcs-web/main.do
Google Chrome 54.0.2840.71 
Mozilla Firefox 49.0.2
Starting Xvfb ...
[2016-11-03 02:16:54] Versions OS: linux 4.4.0-21-generic nodejs: v6.9.1 sitespeed.io: 4.0.2 browsertime: 1.0.0-beta.9 coach: 0.28.4
[2016-11-03 02:16:55] Starting chrome for analysing http://www.ctnma.cn/ioop-bcs-web/main.do 3 time(s)
[2016-11-03 02:16:55] Testing url http://www.ctnma.cn/ioop-bcs-web/main.do run 1
[2016-11-03 02:17:55] Browser failed to start in time, trying one more time.
[2016-11-03 02:18:55] http://www.ctnma.cn/ioop-bcs-web/main.do generated the following error in Browsertime BrowserError: unknown error: Chrome failed to start: crashed
  (Driver info: chromedriver=2.24.417424 (c5c5ea873213ee72e3d0929b47482681555340c3),platform=Linux 4.4.0-21-generic x86_64)
  (Driver info: chromedriver=2.24.417424 (c5c5ea873213ee72e3d0929b47482681555340c3),platform=Linux 4.4.0-21-generic x86_64)
    at BrowsertimeError (/usr/src/app/node_modules/browsertime/lib/support/errors.js:5:5)
    at BrowserError (/usr/src/app/node_modules/browsertime/lib/support/errors.js:13:5)
    at startBrowser.call.catch.tap.tap.catch.catch (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:79:15)
From previous event:
    at SeleniumRunner.start (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:78:13)
    at runIteration (/usr/src/app/node_modules/browsertime/lib/core/engine.js:167:21)
    at Promise.reduce (/usr/src/app/node_modules/browsertime/lib/core/engine.js:242:25)
From previous event:
    at Promise.resolve.tap.tap.tap (/usr/src/app/node_modules/browsertime/lib/core/engine.js:241:17)
From previous event:
    at Engine.run (/usr/src/app/node_modules/browsertime/lib/core/engine.js:240:8)
    at engine.start.then (/usr/src/app/lib/plugins/browsertime/analyzer.js:76:26)
From previous event:
    at Object.analyzeUrl (/usr/src/app/lib/plugins/browsertime/analyzer.js:76:8)
    at Object.processMessage (/usr/src/app/lib/plugins/browsertime/index.js:128:27)
    at queue.process (/usr/src/app/lib/support/queueHandler.js:157:57)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:92:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:73:84)
    at runCallback (timers.js:637:20)
    at tryOnImmediate (timers.js:610:5)
    at processImmediate [as _immediateCallback] (timers.js:582:5)
    at SeleniumRunner.startBrowser (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:37:10)
    at SeleniumRunner.start (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:42:25)
    at runIteration (/usr/src/app/node_modules/browsertime/lib/core/engine.js:167:21)
    at Promise.reduce (/usr/src/app/node_modules/browsertime/lib/core/engine.js:242:25)

**Comment by beenanner** [comment]
@poolwar could you post your script for the auto accept of the alert box? I was able to get everything working by modifying the browsertime seleniumRunner.js to auto accept any alert box.

See: https://github.com/sitespeedio/browsertime/pull/206

**Comment by beenanner** [comment]
I'd prefer to see if we can get it working with the preScript option as I don't like the idea of auto accepting anything that pops up. :-p Also I doubt this will solve multiple alert boxes popping up so the script option would be the more robust solution as it can be customized on a case by case basis.

**Comment by beenanner** [comment]
@poolwar your URL seems to work on my Ubuntu 16.04 machine as well as the docker container.

```
➜  sitespeed.io git:(master) bin/sitespeed.js -b chrome https://www.xiaoying.com
[2016-11-02 23:12:59] Versions OS: linux 4.4.0-45-generic nodejs: v4.4.0 sitespeed.io: 4.0.2 browsertime: 1.0.0-beta.9 coach: 0.28.4
[2016-11-02 23:13:00] Starting chrome for analysing https://www.xiaoying.com 3 time(s)
[2016-11-02 23:13:00] Testing url https://www.xiaoying.com run 1
[2016-11-02 23:13:11] Testing url https://www.xiaoying.com run 2
[2016-11-02 23:13:22] Testing url https://www.xiaoying.com run 3
[2016-11-02 23:13:33] 49 requests, 997.49 kb, firstPaint: 2.74s (±0.22), DOMContentLoaded: 4.59s (±0.35), Load: 7.56s (±0.21), rumSpeedIndex: 5.83s (±0.43) (3 runs)
[2016-11-02 23:13:33] Render HTML for 1 page(s) 
[2016-11-02 23:13:35] HTML stored in /home/beenanner/github/sitespeed.io/sitespeed-result/www.xiaoying.com/2016-11-02-23-12-59
[2016-11-02 23:13:35] Finished analysing https://www.xiaoying.com
```

```
➜  sitespeed.io git:(master) docker run --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io:4.0.2 -b chrome https://www.xiaoying.com                  
Google Chrome 54.0.2840.71 
Mozilla Firefox 49.0.2
Starting Xvfb ...
[2016-11-03 03:14:51] Versions OS: linux 4.4.0-45-generic nodejs: v6.9.1 sitespeed.io: 4.0.2 browsertime: 1.0.0-beta.9 coach: 0.28.4
[2016-11-03 03:14:51] Starting chrome for analysing https://www.xiaoying.com 3 time(s)
[2016-11-03 03:14:51] Testing url https://www.xiaoying.com run 1
[2016-11-03 03:15:00] Testing url https://www.xiaoying.com run 2
[2016-11-03 03:15:11] Testing url https://www.xiaoying.com run 3
[2016-11-03 03:15:21] 48 requests, 996.75 kb, firstPaint: 2.68s (±0.), DOMContentLoaded: 3.72s (±0.18), Load: 6.85s (±0.23), rumSpeedIndex: 4.s (±0.14) (3 runs)
[2016-11-03 03:15:21] Render HTML for 1 page(s) 
[2016-11-03 03:15:23] HTML stored in /sitespeed.io/sitespeed-result/www.xiaoying.com/2016-11-03-03-14-51
[2016-11-03 03:15:23] Finished analysing https://www.xiaoying.com

```

**Comment by kikyoar** [comment]
@beenanner ,thx ,I'll try

**Comment by dustinphipps** [comment]
I _cannot_ confirm that the Docker container works in either 4.0.2 or 4.0.4. Running the same exact command as @beenanner did above, I receive the following output:

```
$ docker run -v $(pwd):/sitespeed.io sitespeedio/sitespeed.io:4.0.2 -b chrome https://www.xiaoying.com
Google Chrome 54.0.2840.71
Mozilla Firefox 49.0.2
Starting Xvfb ...
[2016-11-11 20:33:22] Versions OS: linux 4.8.6-1-ARCH nodejs: v6.9.1 sitespeed.io: 4.0.2 browsertime: 1.0.0-beta.9 coach: 0.28.4
[2016-11-11 20:33:23] Starting chrome for analysing https://www.xiaoying.com 3 time(s)
[2016-11-11 20:33:23] Testing url https://www.xiaoying.com run 1
[2016-11-11 20:34:23] Browser failed to start in time, trying one more time.
[2016-11-11 20:35:23] https://www.xiaoying.com generated the following error in Browsertime BrowserError: unknown error: Chrome failed to start: crashed
  (Driver info: chromedriver=2.24.417424 (c5c5ea873213ee72e3d0929b47482681555340c3),platform=Linux 4.8.6-1-ARCH x86_64)
  (Driver info: chromedriver=2.24.417424 (c5c5ea873213ee72e3d0929b47482681555340c3),platform=Linux 4.8.6-1-ARCH x86_64)
    at BrowsertimeError (/usr/src/app/node_modules/browsertime/lib/support/errors.js:5:5)
    at BrowserError (/usr/src/app/node_modules/browsertime/lib/support/errors.js:13:5)
    at startBrowser.call.catch.tap.tap.catch.catch (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:79:15)
From previous event:
    at SeleniumRunner.start (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:78:13)
    at runIteration (/usr/src/app/node_modules/browsertime/lib/core/engine.js:167:21)
    at Promise.reduce (/usr/src/app/node_modules/browsertime/lib/core/engine.js:242:25)
From previous event:
    at Promise.resolve.tap.tap.tap (/usr/src/app/node_modules/browsertime/lib/core/engine.js:241:17)
From previous event:
    at Engine.run (/usr/src/app/node_modules/browsertime/lib/core/engine.js:240:8)
    at engine.start.then (/usr/src/app/lib/plugins/browsertime/analyzer.js:76:26)
From previous event:
    at Object.analyzeUrl (/usr/src/app/lib/plugins/browsertime/analyzer.js:76:8)
    at Object.processMessage (/usr/src/app/lib/plugins/browsertime/index.js:128:27)
    at queue.process (/usr/src/app/lib/support/queueHandler.js:157:57)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:92:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:73:84)
    at runCallback (timers.js:637:20)
    at tryOnImmediate (timers.js:610:5)
    at processImmediate [as _immediateCallback] (timers.js:582:5)
    at SeleniumRunner.startBrowser (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:37:10)
    at SeleniumRunner.start (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:42:25)
    at runIteration (/usr/src/app/node_modules/browsertime/lib/core/engine.js:167:21)
    at Promise.reduce (/usr/src/app/node_modules/browsertime/lib/core/engine.js:242:25)
[2016-11-11 20:35:23] Render HTML for 1 page(s)
[2016-11-11 20:35:23] HTML stored in /sitespeed.io/sitespeed-result/www.xiaoying.com/2016-11-11-20-33-22
[2016-11-11 20:35:23] Finished analysing https://www.xiaoying.com
```

**Comment by soulgalore** [comment]
Hey @dustinphipps yes I see that you do the same as @beenanner but to get Chrome to work on most platforms you either need to run as *privileged* or with *no-sandbox*: https://www.sitespeed.io/documentation/sitespeed.io/docker/#running-in-docker

Also setting a higher shm-size than default (64mb) will help:
<code>docker run --shm-size=1g --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io -b chrome --browsertime.chrome.args no-sandbox https://www.xiaoying.com</code>

or
<code>docker run --privileged --shm-size=1g --rm -v "$(pwd)":/sitespeed.io sitespeedio/sitespeed.io -b chrome https://www.xiaoying.com</code>
Best
Peter

**Comment by dustinphipps** [comment]
This appears to work, @soulgalore . Thanks for the assistance!

**Comment by dustinphipps** [comment]
@soulgalore Unfortunately, running the same with tsproxy (_i.e._, using the **--connectivity** option) fails every time...

```
Google Chrome 54.0.2840.71
Mozilla Firefox 49.0.2
Starting Xvfb ...
[2016-11-11 23:11:03] Versions OS: linux 4.8.6-1-ARCH nodejs: v6.9.1 sitespeed.io: 4.0.2 browsertime: 1.0.0-beta.9 coach: 0.28.4
[2016-11-11 23:11:04] Starting chrome for analysing https://www.xiaoying.com 3 time(s)
[2016-11-11 23:11:04] Start tsproxy:/usr/src/app/node_modules/browsertime/vendor/tsproxy.py --rtt 200 --inkbps 780 --outkbps 330 -p 1080
[2016-11-11 23:11:04] Testing url https://www.xiaoying.com run 1
[2016-11-11 23:12:06] Could not load URLUrlLoadError: Failed to load https://www.xiaoying.com
[2016-11-11 23:12:06] https://www.xiaoying.com generated the following error in Browsertime UrlLoadError: Failed to load https://www.xiaoying.com
    at BrowsertimeError (/usr/src/app/node_modules/browsertime/lib/support/errors.js:5:5)
    at UrlLoadError (/usr/src/app/node_modules/browsertime/lib/support/errors.js:19:5)
    at getUrl.then.then.catch.catch (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:147:15)
From previous event:
    at SeleniumRunner.loadAndWait (/usr/src/app/node_modules/browsertime/lib/core/seleniumRunner.js:145:13)
    at runner.start.tap.tap.tap (/usr/src/app/node_modules/browsertime/lib/core/engine.js:183:27)
From previous event:
    at runIteration (/usr/src/app/node_modules/browsertime/lib/core/engine.js:183:10)
    at Promise.reduce (/usr/src/app/node_modules/browsertime/lib/core/engine.js:242:25)
From previous event:
    at Promise.resolve.tap.tap.tap (/usr/src/app/node_modules/browsertime/lib/core/engine.js:241:17)
From previous event:
    at Engine.run (/usr/src/app/node_modules/browsertime/lib/core/engine.js:240:8)
    at engine.start.then (/usr/src/app/lib/plugins/browsertime/analyzer.js:76:26)
From previous event:
    at Object.analyzeUrl (/usr/src/app/lib/plugins/browsertime/analyzer.js:76:8)
    at Object.processMessage (/usr/src/app/lib/plugins/browsertime/index.js:128:27)
    at queue.process (/usr/src/app/lib/support/queueHandler.js:157:57)
    at drainItem (/usr/src/app/node_modules/concurrent-queue/index.js:92:21)
    at Immediate.drain (/usr/src/app/node_modules/concurrent-queue/index.js:73:84)
    at runCallback (timers.js:637:20)
    at tryOnImmediate (timers.js:610:5)
    at processImmediate [as _immediateCallback] (timers.js:582:5)
[2016-11-11 23:12:06] Render HTML for 1 page(s)
[2016-11-11 23:12:06] HTML stored in /sitespeed.io/sitespeed-result/www.xiaoying.com/2016-11-11-23-11-03
[2016-11-11 23:12:06] Finished analysing https://www.xiaoying.com
```

**Comment by soulgalore** [comment]
hello @dustinphipps 
try upgrade to the latest 4.0.5 released a couple of minutes ago. The default TSProxy isn't working perfect (for me at least), so could you try our tc implementation? Switch by adding <code>--browsertime.connectivity.engine tc</code>.

Best
Peter

**Comment by tobli** [comment]
I'm closing this since it seems the original issue is resolved. Please open a separate issue for --connectivity if that's still an issue with the latest version. Thanks!

**Comment by dustinphipps** [comment]
@soulgalore Running version 4.0.5 of the container did the trick. I was unable to use the tc engine, but TSProxy works now. Thanks again!

**Comment by beenanner** [comment]
Good to hear you got everything working @dustinphipps! Let us know if you run into anything else.

**Comment by huangyingja** [comment]
Hi,
Seems also got "   " to all test site, may I know how to solve it ?

===================================================================
root@opsw01:~# sitespeed.io http://www.google.com --debug
sleep: using busy loop fallback
[2017-03-16 18:23:02] Versions OS: linux 4.4.0-21-generic nodejs: v7.6.0 sitespeed.io: 4.7.0 browsertime: 1.0.0-beta.31 coach: 0.31.0
[2017-03-16 18:23:03] {

[2017-03-16 18:23:03] Starting chrome for analysing http://www.google.com 3 time(s)
[2017-03-16 18:23:03] Testing url http://www.google.com run 1
[2017-03-16 18:24:03] Browser failed to start in time, trying one more time.

===================================================================

**Comment by soulgalore** [comment]
Hey @huangyingja 
what OS are you using and what version of Chrome? It could be that Chrome and the Chromedriver is out of sync. We love when people use [our Docker container](https://hub.docker.com/r/sitespeedio/sitespeed.io/) since the versions are pinned then.

Best
Peter

**Comment by huangyingja** [comment]
Hi , 
Thanks for your reply. 
OS is 
root@A-NOC-ZABBIX084:~# uname -a
Linux A-NOC-ZABBIX084 4.2.0-27-generic #32~14.04.1-Ubuntu SMP Fri Jan 22 15:32:26 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
Chrome is 
/usr/local/lib
└── chrome@0.1.0
/usr/local/lib
└── chromedriver@2.28.0

**Comment by soulgalore** [comment]
That is not the version for Chrome, that's a starter wrapper version, can you please check? <code>chome --version</code> or something like that?

Do you run headless with XVFB or do you have screen attached? You need to have either for Chrome to work! If you use the Docker container you get all for free :)

Best
Peter

**Comment by huangyingja** [comment]
Hi , 
Thanks for your reply , I will try " https://www.sitespeed.io/3.x/headless/ " first.
Many thanks ~

**Comment by DanielRuf** [comment]
@huangyingja that does not work in the new versions.

**Comment by soulgalore** [comment]
@DanielRuf XVFB works. We dropped support for PhantomJS (since it doesn't support H2, has faulty size values etc). You can run Chrome headless with the new Chrome headless if you use the right parameters, however Chrome Headless doesn't support extensions so some functionality will not work.

**Comment by DanielRuf** [comment]
Right, there is just the prebuilt PhantomJS version. I guess PhantomJS is dead.

Installed XVFB and it worked with Chromium. Would be great to have Chromium headless (native) nevertheless. Unfortunately Firefox, GeckoDriver and PhantomJS are not as smooth as Chromium with XVFB and so on.

I've also installed a Docker image in parallel, this works much better, especially on Debian.

**Comment by soulgalore** [comment]
Ok, but Firefox should work with XVFB, what problem do you see? We have built in support in Browsertime for XVFB, add the --browsertime.xvfb flag and Browsertime handles that (but you need to install it etc). Could be though that some parts is hardcoded just for the setting we use in Docker, let me know if you run into problems so we can fix that.

I tried Chromium headless (native) a month ago and it worked (in Browsertime), it was some problem with setting the size of the viewport (I think the old version in Chromedriver used a extension to do it, but I think that is fixed now). However then we will miss SpeedIndex/FirstVisualChange etc.

Btw to use SpeedIndex etc if you run standalone without Docker you need to install the dependencies for VisualMetrics as in https://github.com/sitespeedio/docker-visualmetrics-deps/blob/master/Dockerfile (but you probably seen that already). I think for Debian that you don't need to use the static version of FFMPEG, but only way is to try, sorry.

**Comment by DanielRuf** [comment]
I did not try VisualMetrics with the non Docker version. I did not test Firefox on the root server. I think it was also another issue with a package dependency and it was not clear enought, that I do not need Selenium and the GeckoDriver (or do I?). Anyway, so far I have working setups.
